### PR TITLE
switched to use the pause image served from gcr

### PR DESCRIPTION
**Pull request by yliaog**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
switched to use the pause image served from gcr

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 1]
```diff
@@ -51,7 +51,7 @@
 #  - Document functions using proper syntax:
 #    https://technet.microsoft.com/en-us/library/hh847834(v=wps.620).aspx
 
-$INFRA_CONTAINER = 'mcr.microsoft.com/k8s/core/pause:1.0.0'
+$INFRA_CONTAINER = 'gcr.io/gke-release/pause-win:1.0.0'
```
Comment (pjh) [inline cluster/gce/windows/k8s-node-setup.psm1]: To clarify: is this GCR bucket available to any user, not just google-owned projects?

Comment (yliaog) [reply cluster/gce/windows/k8s-node-setup.psm1]: Yes, it is available to public.

File: cluster/gce/windows/configure.ps1 [hunk 2]
```diff
@@ -159,6 +158,8 @@ try {
 
   $config = New-FileRotationConfig
   Schedule-LogRotation -Pattern '.*\.log$' -Path ${env:LOGS_DIR} -RepetitionInterval $(New-Timespan -Hour 1) -Config $config
+
+  Pull-InfraContainer
```
Comment (pjh) [inline cluster/gce/windows/configure.ps1]: What's the reason for this change?

On one hand I suppose this should let the node join the cluster more quickly. On the other hand there could be some delay if a user tries to run a Windows pod immediately after the node is Ready but the pause container hasn't finished pulling yet.

Comment (yliaog) [reply cluster/gce/windows/configure.ps1]: it is not needed for kubelet/node to register with the master/control plane. Pull-InfraContainer is a best-effort optimization, so that when a user start a pod, the 'pause' image is likely already present on the node. Moving it to the last would 1) shorten the delay for the node to join the cluster, [so user can submit deployment/pod earlier, user does not need to wait for the node to be ready to submit deployment/pod in theory, but in practice, user may wait/check for node to be ready before deploying anything] 2) no worse than current situation (should be better) in terms of user-perceived node startup time. [we may explore moving it to the first step when making it async/pulling in the background]

Comment (pjh) [reply cluster/gce/windows/configure.ps1]: Makes sense, thanks for clarifying.

Comment (Random-Liu) [reply cluster/gce/windows/configure.ps1]: It doesn't seem safe to move this.

Windows image pull is slow, for docker pulling the pause image takes ~4min, for containerd it is even longer...

If the pause image is not prepulled, and the node becomes ready, it is possible that a user will never be able to create a pod.

The reason is that in kubelet we set a 4min timeout for pod sandbox start (including pulling the pause image) https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/remote/remote_runtime.go#L97.

If the pause image is not prepulled, `RunPodSandbox` may timeout every time, abort the ongoing pulling, and try pulling again. If for any reason the pause image pull is longer than 4min, the image pull may be always aborted, thus the pod can't start any more.

For Docker, I believe this may happen, and for containerd, because of the slow image pull this breaks the test.

Comment (pjh) [reply cluster/gce/windows/configure.ps1]: > If the pause image is not prepulled, RunPodSandbox may timeout every time, abort the ongoing pulling, and try pulling again.

This doesn't sound like the right thing to do on Windows. Can/should we change that behavior?

Comment (yliaog) [reply cluster/gce/windows/configure.ps1]: chatted offline about it.
